packages/util/src/request.ts

```diff
@@ -21,268 +10,55 @@ export enum CLRequestType {
-  sourcePubkey: PrefixedHexString // DATA 48 bytes
-  targetPubkey: PrefixedHexString // DATA 48 bytes
-}
-
 export interface RequestJSON {
-  [CLRequestType.Deposit]: DepositRequestV1
-  [CLRequestType.Withdrawal]: WithdrawalRequestV1
-  [CLRequestType.Consolidation]: ConsolidationRequestV1
+  type: PrefixedHexString
+  data: PrefixedHexString
 }
 
-export type DepositRequestData = {
-  pubkey: Uint8Array
-  withdrawalCredentials: Uint8Array
-  amount: bigint
-  signature: Uint8Array
-  index: bigint
-}
-
-export type WithdrawalRequestData = {
-  sourceAddress: Uint8Array
-  validatorPubkey: Uint8Array
-  amount: bigint
-}
+export class CLRequest<T extends CLRequestType> {
+  // for easy use
+  public readonly bytes: Uint8Array
 
-export type ConsolidationRequestData = {
-  sourceAddress: Uint8Array
-  sourcePubkey: Uint8Array
-  targetPubkey: Uint8Array
-}
-
-export interface RequestData {
-  [CLRequestType.Deposit]: DepositRequestData
-  [CLRequestType.Withdrawal]: WithdrawalRequestData
-  [CLRequestType.Consolidation]: ConsolidationRequestData
-}
-
-export type TypedRequestData = RequestData[CLRequestType]
-
-export interface CLRequestInterface<T extends CLRequestType = CLRequestType> {
-  readonly type: T
-  serialize(): Uint8Array
-  toJSON(): RequestJSON[T]
-}
-
-export abstract class CLRequest<T extends CLRequestType> implements CLRequestInterface<T> {
-  readonly type: T
-  abstract serialize(): Uint8Array
-  abstract toJSON(): RequestJSON[T]
-  constructor(type: T) {
-    this.type = type
+  get type() {
+    return this.bytes[0] as T
   }
-}
 
-export class DepositRequest extends CLRequest<CLRequestType.Deposit> {
-  constructor(
-    public readonly pubkey: Uint8Array,
-    public readonly withdrawalCredentials: Uint8Array,
-    public readonly amount: bigint,
-    public readonly signature: Uint8Array,
-    public readonly index: bigint,
-  ) {
-    super(CLRequestType.Deposit)
+  get data() {
+    return this.bytes.subarray(1)
   }
 
-  serialize() {
-    const indexBytes = this.index === BIGINT_0 ? new Uint8Array() : bigIntToBytes(this.index)
-
-    const amountBytes = this.amount === BIGINT_0 ? new Uint8Array() : bigIntToBytes(this.amount)
-
-    return concatBytes(
-      Uint8Array.from([this.type]),
-      RLP.encode([
-        this.pubkey,
-        this.withdrawalCredentials,
-        amountBytes,
-        this.signature,
-        indexBytes,
-      ]),
-    )
+  constructor(requestType: T, requestData: Uint8Array) {
+    this.bytes = concatBytes(new Uint8Array([requestType]), requestData)
   }
+}
 
-  toJSON(): DepositRequestV1 {
-    return {
-      pubkey: bytesToHex(this.pubkey),
-      withdrawalCredentials: bytesToHex(this.withdrawalCredentials),
-      amount: bigIntToHex(this.amount),
-      signature: bytesToHex(this.signature),
-      index: bigIntToHex(this.index),
-    }
+export class DepositRequest extends CLRequest<CLRequestType.Deposit> {
+  constructor(requestData: Uint8Array) {
```

Comment:
Is it useful to have these sub classes now since they are functionally identical?  Couldn't we just have a `CLRequest` class and then use the `type` property to distinguish between them?